channel/event.go

```diff
@@ -0,0 +1,183 @@
+package channel
+
+import (
+	"log"
+	"slices"
+	"time"
+
+	"github.com/dop251/goja"
+)
+
+type event struct {
+	id         int32
+	duration   time.Duration
+	endTime    time.Time
+	finished   chan struct{}
+	instanceID int
+	playerIDs  []int32
+	server     *Server
+
+	startCallback            func()
+	beforePortalCallback     func(plr scriptPlayerWrapper, src scriptMapWrapper, dst scriptMapWrapper) bool
+	afterPortalCallback      func(plr scriptPlayerWrapper, dst scriptMapWrapper)
+	timeoutCallback          func(plr scriptPlayerWrapper)
+	playerLeaveEventCallback func(plr scriptPlayerWrapper)
+
+	program *goja.Program
+	vm      *goja.Runtime
+}
+
+func createEvent(id int32, instID int, players []int32, server *Server, program *goja.Program) *event {
+	ctrl := &event{
+		id:         id,
+		finished:   make(chan struct{}),
+		instanceID: instID,
+		playerIDs:  players,
+		server:     server,
+		program:    program,
+		vm:         goja.New(),
+	}
+
+	ctrl.vm.SetFieldNameMapper(goja.UncapFieldNameMapper())
+	_ = ctrl.vm.Set("ctrl", ctrl)
+
+	_, err := ctrl.vm.RunProgram(ctrl.program)
+
+	if err != nil {
+		log.Println(err)
```

Comment:
Do we want to just log and continue after failing to run the program? It makes sense to me to return an error instead and stop initialization.

Reply:
Yes, this should return an error